From 9fab4c2b3550e5c6e96e64bf92fa6c25e95c9b3f Thu, 19 Oct 2017 12:03:57 +0200
From: Andrew Branson <andrew.branson@jollamobile.com>
Date: Thu, 19 Oct 2017 12:03:44 +0200
Subject: [PATCH] Disable tracker-remote functionality. 

It is not needed for Sailfish, and requires new dependencies that we cannot easily supply (json-glib, requires Meson).

diff --git a/configure.ac b/configure.ac
index 1f301c7..7ad225d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -298,7 +298,7 @@
                             libxml-2.0     >= $LIBXML2_REQUIRED
                             json-glib-1.0  >= $JSON_GLIB_REQUIRED"
 
-PKG_CHECK_MODULES(LIBTRACKER_REMOTE, [$LIBTRACKER_REMOTE_REQUIRED])
+#PKG_CHECK_MODULES(LIBTRACKER_REMOTE, [$LIBTRACKER_REMOTE_REQUIRED])
 
 # Check requirements for libtracker-extract
 LIBTRACKER_EXTRACT_REQUIRED="glib-2.0     >= $GLIB_REQUIRED
@@ -2660,7 +2660,6 @@
 	src/libtracker-control/tracker-control.pc
 	src/libtracker-data/Makefile
 	src/libtracker-direct/Makefile
-	src/libtracker-remote/Makefile
 	src/libtracker-sparql-backend/Makefile
 	src/libtracker-fts/Makefile
 	src/libtracker-extract/Makefile
diff --git a/src/Makefile.am b/src/Makefile.am
index 532165a..ae3be69 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -11,7 +11,6 @@
 	libtracker-bus                                 \
 	libtracker-data                                \
 	libtracker-direct                              \
-	libtracker-remote                              \
 	libtracker-sparql-backend                      \
 	libtracker-extract                             \
 	libtracker-miner                               \
diff --git a/src/libtracker-sparql-backend/Makefile.am b/src/libtracker-sparql-backend/Makefile.am
index 2f06e09..8c1dfd9 100644
--- a/src/libtracker-sparql-backend/Makefile.am
+++ b/src/libtracker-sparql-backend/Makefile.am
@@ -5,8 +5,7 @@
 	$(BUILD_VALAFLAGS)                             \
 	$(top_srcdir)/src/libtracker-sparql/tracker-sparql-$(TRACKER_API_VERSION).vapi \
 	$(top_srcdir)/src/libtracker-bus/tracker-bus.vapi \
-	$(top_srcdir)/src/libtracker-direct/tracker-direct.vapi \
-	$(top_srcdir)/src/libtracker-remote/tracker-remote.vapi
+	$(top_srcdir)/src/libtracker-direct/tracker-direct.vapi
 
 AM_CPPFLAGS =                                          \
 	$(BUILD_VALACFLAGS)                            \
@@ -22,7 +21,6 @@
 	$(top_builddir)/src/libtracker-sparql/libtracker-sparql.la \
 	$(top_builddir)/src/libtracker-bus/libtracker-bus.la \
 	$(top_builddir)/src/libtracker-direct/libtracker-direct.la \
-	$(top_builddir)/src/libtracker-remote/libtracker-remote.la \
 	$(BUILD_LIBS)                                  \
 	$(LIBTRACKER_REMOTE_LIBS)                      \
 	$(LIBTRACKER_SPARQL_LIBS)
diff --git a/src/libtracker-sparql-backend/tracker-backend.vala b/src/libtracker-sparql-backend/tracker-backend.vala
index 6d7e982..0b8a954 100644
--- a/src/libtracker-sparql-backend/tracker-backend.vala
+++ b/src/libtracker-sparql-backend/tracker-backend.vala
@@ -410,6 +410,3 @@
 	return Tracker.Sparql.Backend.get_internal (cancellable);
 }
 
-public static Tracker.Sparql.Connection tracker_sparql_connection_remote_new (string url_base) {
-	return new Tracker.Remote.Connection (url_base);
-}
diff --git a/src/libtracker-sparql/tracker-connection.vala b/src/libtracker-sparql/tracker-connection.vala
index b6e7fe9..290192f 100644
--- a/src/libtracker-sparql/tracker-connection.vala
+++ b/src/libtracker-sparql/tracker-connection.vala
@@ -218,7 +218,7 @@
 	 *
 	 * Since: 1.12
 	 */
-	public extern static new Connection remote_new (string uri_base);
+	//public extern static new Connection remote_new (string uri_base);
 
 	/**
 	 * tracker_sparql_connection_query:
diff --git a/src/tracker/tracker-sparql.c b/src/tracker/tracker-sparql.c
index 0b4d267..ee14af4 100644
--- a/src/tracker/tracker-sparql.c
+++ b/src/tracker/tracker-sparql.c
@@ -98,7 +98,7 @@
 static gchar *get_shorthand;
 static gchar *get_longhand;
 static gchar *search;
-static gchar *remote_url;
+//static gchar *remote_url;
 
 static GOptionEntry entries[] = {
 	{ "file", 'f', 0, G_OPTION_ARG_FILENAME, &file,
@@ -149,10 +149,10 @@
 	  N_("Returns the full namespace for a class."),
 	  N_("CLASS"),
 	},
-	{ "remote-service", 'r', 0, G_OPTION_ARG_STRING, &remote_url,
-	  N_("Remote service to query to"),
-	  N_("BASE_URL"),
-	},
+//	{ "remote-service", 'r', 0, G_OPTION_ARG_STRING, &remote_url,
+//	  N_("Remote service to query to"),
+//	  N_("BASE_URL"),
+//	},
 	{ NULL }
 };
 
@@ -1075,11 +1075,11 @@
 	TrackerSparqlCursor *cursor;
 	GError *error = NULL;
 
-	if (remote_url != NULL) {
-		connection = tracker_sparql_connection_remote_new (remote_url);
-	} else {
+	//if (remote_url != NULL) {
+	//	connection = tracker_sparql_connection_remote_new (remote_url);
+	//} else {
 		connection = tracker_sparql_connection_get (NULL, &error);
-	}
+	//}
 
 	if (!connection) {
 		g_printerr ("%s: %s\n",

